### PR TITLE
Handle duplicates in aliases

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

If the same alias is mentioned multiple times in a CGD chromosome feature file, each duplicate is added as a new entry ; leading to potential integrity errors.

## Proposal

The list of entries to create should be a unique `Entry` instance. We must remove duplicates _via_ a python `set()`. To do so, each `Entry` instance should have a pre-calculated `pk` that is unique for the combination of an identifier and a repository.

### CODE REVIEW

File: apps/data/io/cgd.py [hunk 1]
```diff
@@ -83,16 +86,36 @@ def _to_entries(self, ignore_aliases):
                 aliases = feature['aliases'].split('|')
 
             for identifier in (feature['name'], *aliases):
+
+                # The Entry primary key is precomputed given an identifier and
+                # a repository. By doing so, we will be able to create a set()
+                # to remove duplicates eventually generated by aliases.
```
Comment (willdurand) [inline apps/data/io/cgd.py]: eventually => potentially

File: apps/data/io/cgd.py [hunk 2]
```diff
@@ -83,16 +86,36 @@ def _to_entries(self, ignore_aliases):
...
+                pk = blake2b(
+                    bytes(
+                        '/'.join((
```
Comment (willdurand) [inline apps/data/io/cgd.py]: why not on a single line?

File: apps/data/tests/io/test_cgd.py [hunk 3]
```diff
@@ -113,6 +120,15 @@ def test__to_entries_with_existing_entries(self):
         )
         assert parser.entries['update'][0].url == first_entry.url
 
+    def test__to_entries_with_aliases(self):
+
+        parser = ChrFeatureParser(self.file_path)
+        parser.parse()
+        parser._to_entries(ignore_aliases=False)
+
+        assert len(parser.entries['new']) == 30
```
Comment (willdurand) [inline apps/data/tests/io/test_cgd.py]: Where does `30` come from?

File: apps/data/io/cgd.py (hunk 2 above)
Comment (jmaupetit) [reply apps/data/io/cgd.py]: At first there was one more argument. But now you are right.

File: apps/data/tests/io/test_cgd.py (hunk 3 above)
Comment (jmaupetit) [reply apps/data/tests/io/test_cgd.py]: 30 is the sum of identifiers (10) and unique aliases (20) in `apps/data/fixtures/C_glabrata_CBS138_current_chromosomal_feature_10.tab`

**Pull request merged**